lcls_tools/common/data_analysis/bmad_modeling/bmad_modeling.py

```diff
@@ -0,0 +1,387 @@
+    match_cmds.append(f"use var {variable}[1:4]")
+    match_cmds.append(f"use dat {datum}[1:4]")
+    tao.cmds(match_cmds)
+    tao.cmd("run")
+
+
+def get_tao(pvdata, mdl_obj):
+    """
+    Returns tao commands from pvdata, if data_source is DES, calls use_klys_when_beam off for Cu Linac
+    """
+    lines_quads, lines_rf = [], []
+    for dm_key, map in mdl_obj.all_data_maps.items():
+        if dm_key.startswith("K"):
+            acc_pv = map.accelerate_pvname
+            if acc_pv == "":
+                continue
+            # acc_pv = f"{acc_pv[0:21]}{mdl_obj.beam_code}{acc_pv[22:]}"
+            # map.accelerate_pvname = acc_pv
+            lines_rf += map.as_tao(pvdata)
+        if dm_key == "cavities":
+            lines_rf = map.as_tao(pvdata)
+        if dm_key == "quads":
+            lines_quads += map.as_tao(pvdata)
+    if "NOT" in mdl_obj.data_source and "cu" in mdl_obj.beam_path:
+        new_lines = []
+        for cmd in lines_rf:
+            if "in_use" in cmd:
+                if "K21_1" in cmd or "K21_2" in cmd:
+                    new_lines.append(cmd)  # L1 always on Beam Code 1
+                    continue
+                ele = cmd.split()[2]
+                [sector, station] = ele[1:].split("_")
+                pv = f"KLYS:LI{sector}:{station}1:"
+                f"BEAMCODE{mdl_obj.beam_code}_STAT"
+                cmd_words = cmd.split()
+                cmd_words[-1] = str(pvdata[pv])
+                new_lines.append(" ".join(cmd_words))
+            else:
+                new_lines.append(cmd)
+        # lines_rf = new_lines
+    return lines_rf + lines_quads
+
+
+def get_machine_values(data_source, pv_list, date_time=''):
+    """Returns pvdata, a dictionary with keys containing the PV name and values from Actual, Desired or Archive"""
+    if data_source in ["DES", "ACT"]:
+        pvdata = get_live(pv_list)
+    elif "ARCHIVE" in data_source:
+        pvdata = lcls_archiver_restore(pv_list, date_time)
+    if "DES" in data_source:
+        for pv, val in pvdata.items():
+            if any(k in pv for k in ['HDSC', 'SWRD', 'STAT']):
+                pvdata[pv] = 0
+            if 'DSTA' in pv:
+                pvdata[pv] = np.array([0, 0])
+    return pvdata
+
+
+def kmod_to_bdes(tao, element, K=0):
+    """Returns BDES given K1 in the Bmad model"""
+    ele = tao.ele_gen_attribs(element)
+    bp = ele["E_TOT"] / 1e9 / 299.792458 * 1e4  # kG m
+    return ele["K1"] * bp * ele["L"]
+
+
+def bdes_to_kmod(tao, element, bdes):
+    """Returns K1 given BDES"""
+    ele = tao.ele_gen_attribs(element)
+    bp = ele["E_TOT"] / 1e9 / 299.792458 * 1e4  # kG m
+    return bdes / ele["L"] / bp  # kG / m / kG m = 1/m^2
+
+
+def get_output(tao):
+    """
+    Returns dictionary of modeled parameters, including element name,
+    twiss and Rmats
+    """
+    yaml_dir = '/sdf/home/c/colocho/lcls-tools/lcls_tools/' \
+               'common/data_analysis/bmad_modeling/yaml/'
```

Comment:
Another place where we can replace the hard coded yaml_dir as in: [lcls_tools/devices/reader.py](https://github.com/slaclab/lcls-tools/blob/f822626ed6c554070354879ba3b27149afeec6e2/lcls_tools/common/devices/reader.py#L12).